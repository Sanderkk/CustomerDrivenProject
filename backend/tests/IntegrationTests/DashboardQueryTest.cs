using System;
using System.Collections.Generic;
using System.IO;
using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using src.Api.Inputs;
using src.Api.Queries;
using src.Api.Types;
using src;
using src.Database;
using src.Database.User;
using Xunit;

namespace tests.IntegrationTests
{
    public class DashboardQueryTest
    {
        private string dbConnectionString;

        public DashboardQueryTest()
        {
            TestDatabaseConfig config = new TestDatabaseConfig();
            dbConnectionString = config.DatabaseConnectionString;
        }
        [Fact]
        public async Task GetDashboard()
        {

            IServiceProvider serviceProvider =
                new ServiceCollection()
                .AddSingleton<IUserRepository, UserRepository>()
                .AddSingleton<IDatabaseConfig>(sp =>
                new DatabaseConfig()
                {
                    DatabaseConnectionString = dbConnectionString
                }
                )
                .BuildServiceProvider();

            IQueryExecutor executor = Schema.Create(c =>
            {
                c.RegisterQueryType(new ObjectType<DashboardQuery>(d => d.Name("Query")));
                c.RegisterType<Dashboard>();
            })
            .MakeExecutable();
            
            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                .SetQuery(@"{allDashboards{ id, name, description, data}}")
                .SetServices(serviceProvider)
                .Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);
            result.MatchSnapshot();
        }

    }
}