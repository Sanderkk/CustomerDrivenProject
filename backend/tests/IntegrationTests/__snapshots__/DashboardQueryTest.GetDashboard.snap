{
  "Data": {
    "allDashboards": [
      {
        "id": 2,
        "name": "Åge",
        "description": "",
        "data": "{\"name\":\"kim\"}"
      },
      {
        "id": 1,
        "name": "Sander",
        "description": "",
        "data": "{\"name\":\"Sander\"}"
      },
      {
        "id": 3,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 4,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 5,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 6,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 7,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 8,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 9,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 10,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 11,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 12,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 13,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 14,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 15,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      },
      {
        "id": 16,
        "name": "Åge",
        "description": "Hei",
        "data": "{\"name\":\"Hei\"}"
      }
    ]
  },
  "Extensions": {},
  "Errors": [],
  "ContextData": {}
}
